plyer/platforms/macosx/filechooser.py

```diff
@@ -75,22 +75,17 @@ def run(self):
         # Mac OS X does not support wildcards unlike the other platforms.
         # This tries to convert wildcards to "extensions" when possible,
         # ans sets the panel to also allow other file types, just to be safe.
-        if len(self.filters) > 0:
+        if self.filters:
             filthies = []
             for f in self.filters:
                 if type(f) == str:
+                    f = (None, f)
+                for s in f[1:]:
                     if not self.use_extensions:
-                        if f.strip().endswith("*"):
+                        if s.strip().endswith("*"):
                             continue
-                        pystr = f.strip().split("*")[-1].split(".")[-1]
+                    pystr = s.strip().split("*")[-1].split(".")[-1]
```

Comment:
this line was previously conditional to `not self.use_extensions` is the change voluntary?

Reply:
Yes, that's actually the main fix to the bug (`pystr` may not be initialized). I believe it was not meant to be indented in the first place, as you could very well have `filters=["*"]`, which is valid... but will crash on MacOS.

The rest of the changes is (mostly) to remove code duplication and to fix it once instead of twice.